### DISCUSSION TITLE
CSVDataset and transform: how to use a column as a parameter of a transform operator?

### DISCUSSION THREAD

**Discussion by yoPitarch** [Project-MONAI/MONAI#7827] [Q&A]
Hello, 

I'd like to use a CSV file to load data and apply transform on these data. 

Before describing the transforms I would like to apply, let me first introduce the structure of my CSV file:

- `img`: relative path to the image
- `label`: relative path to the label
- `x`: x coordinate of the ROI center
- `y`: y coordinate of the ROI center
- `z`: z coordinate of the ROI center

Using `col_groups`, I have transformed `x`, `y` and `z` columns into one column called `center`. Then I want to apply these transforms:
- Load the images and labels
- Ensure first channel
- Resample the spacing to 0.5
- Extract a patch of size 64 and the center is the value of the column `center`

Here is the code I wrote so far to achieve this but I struggle with the "relative ROI center" part since the type of the roi_center parameter must be a Tensor and not a string.

```
csv_dataset = CSVDataset(
    src=df_full,
    col_groups={"center": ["x", "y", "z"]},
    transform=Compose(
        [
            LoadImaged(keys=["img", "label"]),
            EnsureChannelFirstd(keys=["img", "label"]),
            Spacingd(
                keys=["img", "label"],
                pixdim=(0.5, 0.5, 0.5),
                mode=["bilinear", "nearest"],
                padding_mode=["reflection", "zeros"],
                
            ),
            SpatialCropd(keys=["img","label"], roi_size=[64, 64, 64], roi_center="center"),
            EnsureTyped(keys=["img", "label"])
            
        ]
    ),
)
```

Any help would be highly appreciated :)

**Comment by KumoLiu** [comment]
Hi @yoPitarch, the `roi_center` arg in `SpatialCropd` should accept Sequence[int], you may need to load the center first.
You can refer to this tutorial: https://github.com/Project-MONAI/tutorials/blob/ecf8eadbc60277aa8bb604f018e1adf8b75b7dbd/modules/csv_datasets.ipynb#L71

Hope it helps, thanks.
